### DISCUSSION TITLE
Different arming lock for different states.

### DISCUSSION THREAD

**Discussion by RaffoNin** [MiguelRipoll23/homebridge-securitysystem#153] [Ideas]
I am loving the arming lock functionality and it has worked flawlessly so far for me despite it being in beta. I would love different arming locks per different states though as states like home and away have different triggers (away has more contact sensors).

**Comment by MiguelRipoll23** [comment]
Hey there,

Great to hear that, could you tell me how are you using it? I really need it to know as depending of your setup, it may be best suited for you to reset the switch depending of the mode set or you will run into some big desynchronization between all the arming lock switches.

**Reply by RaffoNin** [reply]
I currently have an automation wherein the arming lock turns on when one of my contact sensors are opened and it turns off when a contact sensor is closed under the condition that the other contact sensor is closed. 

The problem is the contact sensor that I need to trigger are different for home and away. For example ,away mode is bound to more contact sensors. I would love two types of arming locks per state because I want to be able to set an "arming lock for away" that has an automation that won't allow the away mode to be triggered when any of my 20 contact sensors are open. While in the home mode I want a "arming lock for home" that has an automation that wont allow home mode to be triggered when only 10 of my contact sensors are opened.

**Reply by MiguelRipoll23** [reply]
Oh, I see. How would you feel if I replace that global switch with three individual arming lock switches for each mode in order to remove unnecessary switches?

It would be informed on the next update and then removed, alongside the new ones for each mode.

**Reply by RaffoNin** [reply]
That would be just what I need. As long as it's possible to do of course. Thank you.

**Reply by MiguelRipoll23** [reply]
It's now available in the beta channel for testing, please keep the discussion going if needed in the following issue #156. Thanks for the feedback!

**Reply by MiguelRipoll23** [reply]
It's live in the stable channel, enjoy!

**Reply by RaffoNin** [reply]
Just tested it right now with the automations I had in mind. Works perfectly so far. Appreciate the work on the plug-in!
